src/main/java/seedu/address/model/ItemModelManager.java

```diff
@@ -460,6 +461,16 @@ public void sort() {
         this.visualList = visualList.sort();
     }
 
+    /**
+     * Sorts the current visual list based on a comparator.
+     * @param comparator the comparator to sort the current list by.
+     */
+    public void sort(Comparator<Item> comparator) {
+        VisualizeList tempList = visualList.deepCopy();
```

Comment:
is there a need to deepcopy, sort and then reassign? can you sort the visual list directly?

Reply:
Hi, the reason why I do this is because I want to keep the underlying list in it's order in which it is added as there is no way to bring it back to it's order when added. As such, I will create a new list every time and return that list.

Reply:
alright got it!